src/main/kotlin/exercise3/task3/SherlockValidatesString.kt

```diff
@@ -29,7 +29,19 @@ package exercise3.task3
  */
 
 internal fun isSherlockValid(s: String): String {
-    TODO("Implement me!!!")
+    require(s.length in 1..100000) { "String length must be in range from 1 to 10^5" }
+    require(s.all{ it in 'a'..'z' }) { "String must only contain small letters (a..z)" }
+
+    val occurrences = s.groupingBy { it }.eachCount().map { it.value }
+    val cMin = occurrences.min()
+    val cMax = occurrences.max()
+
+    return when {
+        cMin == cMax -> "YES"
+        cMax - cMin == 1 && occurrences.count { it == cMax } == 1 -> "YES"
+        cMin == 1 && occurrences.count { it == cMin } == 1 && occurrences.count { it == cMax } == occurrences.size - 1 -> "YES"
```

Comment:
If I'm not mistaken, your solution isn't fully correct. 
for the test case `as`, your function would return `YES` even though the expected result should be `NO`.